resources/services/resourcegroups/resource_groups.go

```diff
@@ -0,0 +1,153 @@
+				Name:        "resource_query",
+				Description: "The query that defines a group or a search.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "arn",
+				Description: "The ARN of the resource group",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("GroupArn"),
+			},
+			{
+				Name:        "name",
+				Description: "The name of the resource group",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "description",
+				Description: "The description of the resource group",
+				Type:        schema.TypeString,
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+
+func fetchResourcegroupsResourceGroups(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	return diag.WrapError(client.ListAndDetailResolver(ctx, meta, res, listResourceGroups, resourceGroupDetail))
+}
+
+func listResourceGroups(ctx context.Context, meta schema.ClientMeta, detailChan chan<- interface{}) error {
+	var config resourcegroups.ListGroupsInput
+	c := meta.(*client.Client)
+	svc := c.Services().ResourceGroups
+	for {
+		output, err := svc.ListGroups(ctx, &config, func(options *resourcegroups.Options) {
+			options.Region = c.Region
+		})
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		for _, item := range output.GroupIdentifiers {
+			detailChan <- item.GroupArn
+		}
+		if aws.ToString(output.NextToken) == "" {
+			break
+		}
+		config.NextToken = output.NextToken
+	}
+	return nil
+}
+
+func resourceGroupDetail(ctx context.Context, meta schema.ClientMeta, resultsChan chan<- interface{}, errorChan chan<- error, listInfo interface{}) {
+	c := meta.(*client.Client)
+	groupArn := listInfo.(*string)
+	svc := c.Services().ResourceGroups
+	groupResponse, err := svc.GetGroup(ctx, &resourcegroups.GetGroupInput{
+		Group: groupArn,
+	})
+	if err != nil {
+		if c.IsNotFoundError(err) {
+			return
+		}
+		errorChan <- diag.WrapError(err)
+		return
+	}
+	resultsChan <- groupResponse.Group
+}
+
+func resolveGroupQuery(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource) error {
+	group := resource.Item.(*types.Group)
+	cl := meta.(*client.Client)
+	svc := cl.Services().ResourceGroups
+	input := resourcegroups.GetGroupQueryInput{
+		Group: group.GroupArn,
+	}
+	output, err := svc.GetGroupQuery(ctx, &input, func(options *resourcegroups.Options) {
+		options.Region = cl.Region
+	})
+	if err != nil {
+		return diag.WrapError(err)
+	}
+	if err := resource.Set("resource_query_type", output.GroupQuery.ResourceQuery.Type); err != nil {
+		return diag.WrapError(err)
+	}
+	return diag.WrapError(resource.Set("resource_query", output.GroupQuery.ResourceQuery.Query))
+}
```

Comment:
I think this can be deleted

Reply:
As in you don't want to capture this information? GetGroup doesn't return the query

Reply:
Or just bundle that call with `resourceGroupDetail`?

Reply:
Exactly. Typically we only use `PostResourceResolver` as last resort. I opened[ a pr](https://github.com/cloudquery/cq-provider-aws/pull/1398) showing how you can  simplify your resource and not need to use a `PostResourceResolver`.  Once you see the changes I made and incorporate them into this PR I will close my pr and we can merge yours!

Reply:
Feel free to ping me on Discord if you have any questions!

Reply:
Well would never have guessed that you are new to Go, you're doing a great job!